docs/SIL.rst

```diff
@@ -2608,6 +2682,76 @@ undefined behavior if the global variable has already been initialized.
 
 The type operand must be a lowered object type.
 
+begin_async_continuation
+````````````````````````
+
+::
+
+  sil-instruction ::= 'begin_async_continuation' '[throws]'? sil-type
+
+  %0 = begin_async_continuation $T
```

Comment:
@jckarter to maintain future flexibility, did you consider making this have potentially multiple instruction results? Not saying it has it today, but if we want to add something this will ensure that we do not run into problems ala apply site.

Reply:
`begin_async_continuation` doesn't traffic in user-provided values and so doesn't need multiple results.  `await_async_continuation` *does* traffic in user-provided values, but I think the usefulness of producing separated values is probably tied to the usefulness of allowing reabstraction of the continuation result — which is to say, per the conversion I had above with Joe, not very useful.

Reply:
Maybe it isn't useful today, but we should be cognizant that requirements can change in the future. It is easy to treat a multiple result instruction as a single result instruction. Hard to go the other way due to source incompatibility. That being said if we are /really/ sure there is no case in the future where that will be needed. I am cool. Just be aware that we are boxing in the design and potentially (if requirements change) requiring painful refactoring.

Reply:
How hard would it be to add multiple results later if we decide it's useful? For now, YAGNI seems prudent.